lib/ContractInteract/PricerRule.js

```diff
@@ -0,0 +1,348 @@
+
+    if (!this.contract) {
+      const err = new TypeError(`Could not load PricerRule contract for: ${this.address}`);
+      return Promise.reject(err);
+    }
+  }
+
+  /**
+   * Deploys PricerRule contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Origin chain web3 object.
+   * @param {String} organization Organization address.
+   * @param {String} eip20Token The economy token address.
+   * @param {String} tokenRules The economy token rules address.
+   * @param {String} baseCurrencyCode The economy base currency code.
+   * @param {String} conversionRate The conversion rate from the economy base currency
+   *                        to the token. e.g. CR of "OST => Unsplash"
+   * @param {String} conversionRateDecimals The conversion rate's decimals from the
+   *                                economy base currency to the token.
+   * @param {String} requiredPriceOracleDecimals Required decimals for price oracles.
+   * @param {Object} txOptions Tx options.
+   *
+   * @returns {Promise<PricerRule>} Promise containing the PricerRule
+   *                              instance that has been deployed.
+   */
+  static async deploy(
+    auxiliaryWeb3,
+    organization,
+    eip20Token,
+    tokenRules,
+    baseCurrencyCode,
+    conversionRate,
+    conversionRateDecimals,
+    requiredPriceOracleDecimals,
+    txOptions
+  ) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address: ${txOptions.from}.`);
+      return Promise.reject(err);
+    }
+
+    const tx = PricerRule.deployRawTx(
+      auxiliaryWeb3,
+      organization,
+      eip20Token,
+      tokenRules,
+      baseCurrencyCode,
+      conversionRate,
+      conversionRateDecimals,
+      requiredPriceOracleDecimals
+    );
+
+    return Utils.sendTransaction(tx, txOptions).then((txReceipt) => {
+      const address = txReceipt.contractAddress;
+      return new PricerRule(auxiliaryWeb3, address);
+    });
+  }
+
+  /**
+   * Method which returns Tx object to deploy GnosisSafe master copy contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Auxiliary chain web3 object.
+   * @param {String} organization Organization address.
+   * @param {String} eip20Token The economy token address.
+   * @param {String} tokenRules The economy token rules address.
+   * @param {String} baseCurrencyCode The economy base currency code.
+   * @param {String} conversionRate The conversion rate from the economy base currency
+   *                        to the token.
+   * @param {String} conversionRateDecimals The conversion rate's decimals from the
+   *                                economy base currency to the token.
+   * @param {String} requiredPriceOracleDecimals Required decimals for price oracles.
+   *
+   * @returns {Object} Raw transaction.
+   */
+  static deployRawTx(
```

Comment:
any validation on the params required?

Reply:
Added more validations on the addresses.